nsa/train.py

```
@@ -423,64 +533,79 @@ def train(args):
+    print(f"\nCheckpoint saved to: {args.checkpoint_dir}/best_nsa_{args.model_size}_model.pth")
+
+    # ==========================================================================
+    # MLflow Finalization
+    # ==========================================================================
+
+    if use_mlflow:
+        # Log final best IoU metric
+        mlflow.log_metric("best_iou", best_iou)
+        # End the MLflow run
+        mlflow.end_run()
+        print(f"\nMLflow run completed successfully.")
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)